File: app/src/main/java/com/mlkit/textrecognition/app/di/components/ActivityModule.java
package com.mlkit.textrecognition.app.di.components;

import com.mlkit.textrecognition.presentation.modules.main.MainActivity;
import com.mlkit.textrecognition.presentation.modules.splash.SplashActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityModule {

    @ContributesAndroidInjector
    abstract MainActivity mainActivity();

    @ContributesAndroidInjector
    abstract SplashActivity splashActivity();
}

File: app/src/main/java/com/mlkit/textrecognition/app/di/components/ActivityModule.kt
package com.mlkit.textrecognition.app.di.components

import com.mlkit.textrecognition.presentation.modules.main.MainActivity
import com.mlkit.textrecognition.presentation.modules.splash.SplashActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityModule {

    @ContributesAndroidInjector
    internal abstract fun mainActivity(): MainActivity

    @ContributesAndroidInjector
    internal abstract fun splashActivity(): SplashActivity
}
